extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisSequenceNumber.java

```diff
@@ -62,7 +62,9 @@
   private KinesisSequenceNumber(String sequenceNumber, boolean isExclusive)
   {
     super(sequenceNumber, isExclusive);
-    if (END_OF_SHARD_MARKER.equals(sequenceNumber) || NO_END_SEQUENCE_NUMBER.equals(sequenceNumber)) {
+    if (END_OF_SHARD_MARKER.equals(sequenceNumber)
+        || NO_END_SEQUENCE_NUMBER.equals(sequenceNumber)
+        || EXPIRED_MARKER.equals(sequenceNumber)) {
```

Comment:
nit: use `isValidAWSKinesisSequence` instead
```suggestion
    if (!isValidAWSKinesisSequence(sequenceNumber)) {
```

Reply:
I avoided that deliberately since we may have a sequence number that does not indicate max sequence but is not a valid kinesis sequence number either.

Reply:
makes sense 👍